server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -623,11 +624,42 @@ public boolean configure(final String name, final Map<String, Object> params) th
 
         _allowSubdomainNetworkAccess = Boolean.valueOf(_configs.get(Config.SubDomainNetworkAccess.key()));
 
+        verifyConfigDriveEntriesOnZones();
+
         s_logger.info("Network Service is configured.");
 
         return true;
     }
 
+    /**
+     * Verifies ConfigDrive entries on a zone. Adds Diabled ConfigDrive provider if missing
+     */
+    private void checkConfigDriveEntriesOnZone(DataCenterVO zone) {
```

Comment:
What do you think of changing the name from `checkConfigDriveEntriesOnZone` to `addDisabledConfigDriveEntriesOnZone`?

Reply:
would you mind adding some unit tests there as well?

Reply:
Done, thanks